### PR TITLE
[lldb][test] TestDataFormatterLibcxxStringSimulator.py: add new padding layout

### PR DESCRIPTION
**Pull request by Michael137**
Depends on https://github.com/llvm/llvm-project/pull/108362 and https://github.com/llvm/llvm-project/pull/108343.

Adds new layout for https://github.com/llvm/llvm-project/pull/105865.

### CODE REVIEW

File: lldb/test/API/functionalities/data-formatter/data-formatter-stl/libcxx-simulators/string/main.cpp [hunk 1]
```diff
@@ -77,7 +89,12 @@ template <class _CharT, class _Traits, class _Allocator> class basic_string {
     };
 #else // !SUBCLASS_PADDING
 
+#ifdef UB_PADDING
     unsigned char __padding[sizeof(value_type) - 1];
```
Comment (labath) [inline lldb/test/API/functionalities/data-formatter/data-formatter-stl/libcxx-simulators/string/main.cpp:93]: Was that (the zero sized array) really UB?  Or just a compiler extension?

**Review by labath — APPROVED** [review]

File: lldb/test/API/functionalities/data-formatter/data-formatter-stl/libcxx-simulators/string/main.cpp (hunk 1 above)
Comment (Michael137) [reply lldb/test/API/functionalities/data-formatter/data-formatter-stl/libcxx-simulators/string/main.cpp:93]: Good point! I'll rename it to `NON_STANDARD_PADDING`

**Comment by Michael137** [comment]
(this libc++-side change was relanded in: https://github.com/llvm/llvm-project/pull/108867)

**Comment by Michael137** [comment]
Broke the [lldb-aarch64-windows](https://lab.llvm.org/buildbot/#/builders/141)[2921](https://lab.llvm.org/buildbot/#/builders/141/builds/2921) bot

**Pull request merged**